src/pointarray.cpp

```diff
@@ -61,6 +61,48 @@ struct OctreeNode
             delete children[i];
     }
 
+    bool rayPassesNearOrThrough(
+        const V3d& rayOrigin,
+        const V3d& rayDir) const
+    {
+        const double diagRadius = 1.2*bbox.size().length()/2; // Sphere diameter is length of box diagonal
+
+        V3d o2c = bbox.center() - rayOrigin; // vector from rayOrigin to box center
+
+        const double l = o2c.length();
+
+        if(l < diagRadius)
+            return true; // rayOrigin lies within bounding sphere
+
+        // rayOrigin lies outside of bounding sphere
+
+        const double cosA = (o2c ^ rayDir)/l;  // cosine of angle between rayDir and vector from origin to center
+
+        if(cosA < DBL_MIN)
+            return false; // rayDir points to side or behind with respect to direction from origin to center of box
+
+        const double sinA = sqrt(1 - cosA*cosA); // sine of angle between rayDir and vector from origin to center
+
+        return sinA/cosA < diagRadius/l;
+    }
+
+    size_t closestPoint(
+        const V3f *const p,
+        const V3f& rayOrigin,
+        const V3f& rayDirection,
+        const double longitudinalScale,
+        double& thisRClosest) const
+    {
+         // calls utils closestPointToRay for the points within this octree node
+         return beginIndex + closestPointToRay(
+                                 p + beginIndex,
+                                 endIndex - beginIndex,
+                                 rayOrigin,
+                                 rayDirection,
+                                 longitudinalScale,
+                                 &thisRClosest);
```

Comment:
This is neater, thanks.

Again with the function arguments, I prefer compact code where possible (function arguments on one line), but readability counts so I'll leave that up to your judgment here.

For formatting consistency, if you can indent relative to the opening brace that would be helpful, thus:

```C++
return beginIndex + closestPointToRay(p + beginIndex,
                                      endIndex - beginIndex,
                                      rayOrigin,
                                      rayDirection,
                                      longitudinalScale,
                                      &thisRClosest);
```